### PR TITLE
Trim MS4525D0 I2C airspeed sensor init function

### PR DESCRIPTION
**Pull request by tylerfaye**
Changed "bool AP_Airspeed_MS4525::init()" so that it would more closely match the style of the other Airspeed sensor initialization functions, as well as added comments for readability.

Specifically used the "get_bus()" function instead of a for loop and array to probe each bus for the sensor one by one. No need to have the loop in the init function when there's an already available function to do the work for you.

Related to #9715 and referenced in the comments there as well.

### CODE REVIEW

**Review by khancyr — COMMENTED** [review]
@tridge @WickedShell from my reading of https://github.com/ArduPilot/ardupilot/issues/9715 the initialization of the MS4525 is done by default by looking at all available port. We can argue about refactoring that in a better way for easier maitenance but that isn't the point.

This PR (that isn't compiling) proposes a breaking change that request user to select the right bus for MS4525. It can be easily rework (and save 60bytes on cubeblack build) but bring a breaking changes. Should I make the corrections or we close this ?

**Comment by tridge** [comment]
the reason the MS4525 scans all buses is it can't be confused on the bus with another sensor type. The MS5525 looks the same as a MS5611 baro, so we can't scan all buses (which is why we added the bus parameter)
I don't think we should change this as it is a breaking change for our most widely used airspeed sensor

**Comment by WickedShell** [comment]
@tridge that's actually not true, MS4525 looks the same as a DLVR, and it's currently completely incompatible unless you know the bus scan order. To make it worse if it finds a DLVR it's almost correct, but isn't correct.

Also unless I've forgotten something (I haven't looked), isn't this completely incompatible with having 2 MS4525 sensors, as it will always scan in the same order? And this is really non obvious to users, as it looks like they'd have 2 sensors. But I haven't double checked this, this was just a recollection from memor.y

I'd love to see this changed to stop scanning everything, and have considered it in the past, but it's problematic without breaking setups.

File: libraries/AP_Airspeed/AP_Airspeed_MS4525.cpp [hunk 1]
```diff
@@ -34,44 +34,45 @@ AP_Airspeed_MS4525::AP_Airspeed_MS4525(AP_Airspeed &_frontend, uint8_t _instance
...
+
+    // Report if sensor was not found, return false if initialization unsuccessful
     if (!found) {
         printf("MS4525: no sensor found\n");
         return false;
     }
 
+    // Report I2C bus and address if the sensor was found, if statement not strictly needed
+    if (found) {
+        printf("MS4525: Found sensor on bus %u address 0x%02x\n", get_bus(), MS4525D0_I2C_ADDR);
```
Comment (rmackay9) [inline libraries/AP_Airspeed/AP_Airspeed_MS4525.cpp:73]: We shouldn't try and avoid output when everything is working fine.

**Comment by tridge** [comment]
doesn't build .... plus breaks existing setups
I think we'd need a new airspeed type, MS4525_BUS which does honour the bus param

**Comment by khancyr** [comment]
as stated in comments this need an heavy rework to no break the current setup.
I close this PR and an issue as been opened to keep track of this : https://github.com/ArduPilot/ardupilot/issues/15855
Thanks for the contribution, feel free to reopen a PR with the changes asked.

**Pull request closed without merging**